File: leetcode/1929_concatenation_of_array/leetcode1929.js
/**
 * Tags: Arrays, Simulation
 * Difficulty: Easy
 *
 * Approach: Bruteforce
 * Time complexity: O(N)
 * Space complexity: O(1)
 */

function solution1(nums) {
	let ans = [];
	const n = nums.length;

	for (let i = 0; i < n; i++) {
		ans[i] = nums[i];
		ans[i + n] = nums[i];
	}

	return ans;
}

/**
 * Approach: Mathematical approach using modulus
 * Time complexity: O(N)
 * Space complexity: O(1)
 */

function solution2(nums) {
	let ans = [];
	for (let i = 0; i < nums.length * 2; i++) {
		ans.push(nums[i % nums.length]);
	}
	return ans;
}

/**
 * Approach: Using built-in concat()
 * Time complexity: O(N+M)
 * Space complexity: O(N+M)
 */

function solution3(nums) {
	return nums.concat(nums);
}

File: leetcode/Leetcode1929.java
package leetcode;

public class Leetcode1929 {

    /**
     * Approach: Bruteforce
     * Time complexity: O(N)
     * Space complexity: O(1)
     */

    public int[] solution1(int[] nums) {
        int n = nums.length;
        int[] result = new int[2 * n];

        for (int i = 0; i < nums.length; i++) {
            result[i] = nums[i];
            result[i + n] = nums[i];
        }

        return result;
    }

    /**
     * Approach: Mathematical approach using modulus
     * Time complexity: O(N)
     * Space complexity: O(1)
     */
    public int[] solution2(int[] nums) {
        int n = nums.length;
        int[] result = new int[2 * n];

        for (int i = 0; i < n; i++) {
            result[i] = nums[i % n];
        }

        return result;
    }
}
